src/server.rs

```diff
@@ -230,8 +234,16 @@ where
                         self.config.connect_rule(),
                         self.tx_cmd.clone(),
                     );
-                    self.session
-                        .insert(session.id, spawn_session(session, tx, addr, stream));
+                    let id = session.id;
+                    match spawn_session(session, tx, addr, stream) {
+                        Ok(th) => {
+                            self.session.insert(id, th);
+                        }
+                        Err(err) => {
+                            error!("spawn_session: {}", err);
+                            debug!("spawn_session: {:?}", err);
+                        }
```

Comment:
If spawn_session fails with an error the mainloop continues running just ignoring the error.
Is this behavior the expected one?

Reply:
This is pointless and has been removed.